onnxruntime/core/platform/windows/env.cc

```diff
@@ -59,18 +59,33 @@ class WindowsThread : public EnvThread {
  public:
   WindowsThread(const ORTCHAR_T* name_prefix, int index,
                 unsigned (*start_address)(int id, Eigen::ThreadPoolInterface* param), Eigen::ThreadPoolInterface* param,
-                const ThreadOptions& thread_options)
-      : hThread((HANDLE)_beginthreadex(nullptr, thread_options.stack_size, ThreadMain,
-                                       new Param{name_prefix, index, start_address, param, thread_options}, 0,
-                                       &threadID)) {
+                const ThreadOptions& thread_options) {
+    custom_create_thread_fn = thread_options.custom_create_thread_fn;
+    custom_thread_creation_options = thread_options.custom_thread_creation_options;
+    custom_join_thread_fn = thread_options.custom_join_thread_fn;
+
+    if (custom_create_thread_fn) {
+      custom_thread_handle = custom_create_thread_fn(custom_thread_creation_options, (OrtThreadWorkerFn)CustomThreadMain, new Param{name_prefix, index, start_address, param, thread_options});
```

Comment:
Please use C++ style casts. Several places.

Reply:
Did changes in test_inference.cc and test_main.cc with c++ sytle cast.